units/provisioning_setup_host.go

```diff
@@ -321,7 +321,7 @@ func (j *setupHostJob) putJasperCredentials(ctx context.Context, fileName string
 		return errors.Wrap(err, "could not export Jasper credentials")
 	}
 
-	file, err := ioutil.TempFile("", fileName)
+	file, err := ioutil.TempFile("", filepath.Base(fileName))
```

Comment:
What's the reason for calling `filepath.Base` here?

Reply:
`ioutil.TempFile()` relies on the `$TMP` dir to store the file (`/tmp`). However, it won't recursively make directories for you in the temp dir, so if you give it a file path instead of a just the base file name (in this case `~/setup.sh`), it will error with no such file or directory because `/tmp/~` is not an existing directory.